src/bin/repl/mod.rs

```diff
@@ -93,7 +98,92 @@ impl Repl {
                 None => println!("unknown command '{}'", name),
             }
         } else {
-            todo!()
+            match self.execute_code(line) {
+                Ok(_) => {}
+                Err(err) => {
+                    // TODO: Proper error reporting
+                    println!("error: {}", err.data);
+                }
+            }
         }
     }
+
+    fn execute_code(&mut self, code: &str) -> Result<(), CompileError> {
+        let module = initialize_jit_module();
+
+        let expr = analyze_expr(code)?;
+        let expr_ty = expr.ctype.clone();
+        let decl = wrap_expr(expr);
+        let module = ir::compile(module, vec![decl], false).0?;
+
+        let mut jit = JIT::from(module);
+        jit.finalize();
+        let execute_fun = jit
+            .get_compiled_function("execute")
+            .expect("this is not good.");
+
+        match expr_ty {
+            Type::Long(signed) => {
+                let result = unsafe {
+                    let execute: unsafe extern "C" fn() -> u64 = std::mem::transmute(execute_fun);
```

Comment:
Any reason you can't return `i64` directly here? They're represented the same at the hardware level.
```suggestion
                    let execute: unsafe extern "C" fn() -> i64 = std::mem::transmute(execute_fun);
```

Reply:
Yes I can do that, but then I still have to cast it to a `u64` if it's unsigned, so there's not really a difference to the current code